### PR TITLE
BrightIdWidget

### PR DESCRIPTION
**Pull request by corwintines**
Completes getter functionality for BrightIdWidget component for its use cases in the CallToAction and Profile components.

https://www.notion.so/efdn/Build-BrightIdWidget-component-869e1356fd0549eeb3f21ea4c13124d5
https://www.notion.so/efdn/Build-CallToActionCard-component-6682b695bbbb46d59609d4f53d62270f

### CODE REVIEW

File: vue-app/src/components/BrightIdWidget.vue [hunk 1]
```diff
@@ -1,21 +1,20 @@
 <template>
   <div v-if="isProjectCard" class="setup-container-project-card">
     <div class="row">
-      <div v-if="isVerified">
+      <div v-if="isLinked && isRegistered && isSponsored && isVerified">
```
Comment (pettinarip) [inline vue-app/src/components/BrightIdWidget.vue]: We can just check the `isRegistered` flag. If that flag is `true` that means the others are `true` as well.

If the user was added to the registry that means that the user has been verified before.

File: vue-app/src/components/CallToActionCard.vue [hunk 2]
```diff
@@ -58,24 +58,23 @@ import { userRegistryType, UserRegistryType } from '@/api/core'
...
+      this.$store.state.currentUser.brightId.isLinked
     )
   }
 
-  get isUserUnique(): boolean {
+  get isUserVerified(): boolean {
     return (
-      this.$store.state.currentUser && this.$store.state.currentUser.isUnique
+      this.$store.state.currentUser &&
+      this.$store.state.currentUser.brightId.isVerified
```
Comment (samajammin) [inline vue-app/src/components/CallToActionCard.vue]: `isVerified` is now on `currentUser`

**Comment by samajammin** [comment]
@corwintines I feel like clicking "start setup" in the BrightIdWidget should close the Profile component - what do you think?

Feels weird how the page navigates but it stays open with the overlay:
Image 2021-08-09 at 5 55 41 PM

File: vue-app/src/components/BrightIdWidget.vue [hunk 3]
```diff
@@ -1,21 +1,20 @@
 <template>
   <div v-if="isProjectCard" class="setup-container-project-card">
```
Comment (samajammin) [inline vue-app/src/components/BrightIdWidget.vue]: This code is not very DRY - most of this is entirely re-used when `isProjectCard` is `false`. Let's refactor this a bit.

Comment (samajammin) [reply vue-app/src/components/BrightIdWidget.vue]: See what I mean?
Image 2021-08-09 at 6 02 21 PM
Seems the only difference is when this widget is in the Profile (i.e. `!isProjectCard`), it also has that lower row (highlighted in red)

**Review by samajammin — CHANGES_REQUESTED** [review]
A couple comments to address.

Note: I pushed a small commit, so pull before making changes 👍

File: vue-app/src/components/BrightIdWidget.vue [hunk 4]
```diff
@@ -43,21 +42,20 @@
...
       </div>
-      <div v-if="!isVerified">
+      <div v-else>
         <icon-status
           v-bind:sad="true"
           logo="brightid.svg"
           secondaryLogo="close-black.svg"
         />
       </div>
-      <!-- TODO: should this icon demo verified status or completion of entire setup? Is it even needed?-->
```
Comment (samajammin) [inline vue-app/src/components/BrightIdWidget.vue:60]: (I know you didn't implement this) this class logic below is a bit sloppy:
```
      <p>
        <span
          :class="{
            step0: !isLinked && !isVerified && !isSponsored && !isRegistered,
            step1: isLinked,
            step2: isVerified || isSponsored,
            step3: isVerified && isSponsored,
            step4: isRegistered,
          }"
        />
        / 4
      </p>
```
We don't need CSS here, we should just calculate the number step we're at. @pettinarip implemented this basic logic here: https://github.com/ethereum/clrfund/blob/brightid-integration-2/vue-app/src/views/Verify.vue#L326

Then we can just display {{currentStep +1}} in the UI.

File: vue-app/src/components/CallToActionCard.vue (hunk 2 above)
Comment (corwintines) [reply vue-app/src/components/CallToActionCard.vue]: Isn't this still in brightId?

Comment (pettinarip) [reply vue-app/src/components/CallToActionCard.vue]: hmm yes, this is indeed correct, is in `currentUser.brightId.isVerified`. I think Sam got confused by the `isRegistered` flag that as he said lives in `currentUser`.

File: vue-app/src/components/BrightIdWidget.vue (hunk 4 above)
Comment (corwintines) [reply vue-app/src/components/BrightIdWidget.vue:60]: Sounds good 👍

File: vue-app/src/components/BrightIdWidget.vue (hunk 3 above)
Comment (corwintines) [reply vue-app/src/components/BrightIdWidget.vue]: Good point, will take a towel to this and DRY it up

**Review by pettinarip — APPROVED** [review]
Overall this looks great!

I have one small comment to see what you think. When you click on "Continue setup", you get transitioned to the `/verify` page. I have the feeling that we should auto close the `Profile` panel too.

**Comment by corwintines** [comment]
@pettinarip yup should have been there, my bad. Quickly added it back.

**Review by samajammin — APPROVED** [review]
Nice work! I cleaned up some styles & logic. Merging 👍 👍

**Pull request merged**